### PR TITLE
Flexmeasures documentation

### PR DESCRIPTION
**Pull request by GustaafL**
## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [ ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/flexmeasures.markdown [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+soc_sensor:
+  description: The state of charge sensor of the flexible energy asset.
+  required: true
+  type: integer
+rm_discharge_sensor:
+  description: The resource manager discharge sensor.
+  required: true
+  type: integer
+schedule_duration:
+  description: The duration for which the schedules should be calculated in hours.
```
Comment (victorgarcia98) [inline source/_integrations/flexmeasures.markdown]: This should be `The duration of the schedule in ISO8601 format.`

File: source/_integrations/flexmeasures.markdown [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+  required: true
+  default: 24
```
Comment (victorgarcia98) [inline source/_integrations/flexmeasures.markdown]: "PT24H"

File: source/_integrations/flexmeasures.markdown [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+  type: integer
```
Comment (victorgarcia98) [inline source/_integrations/flexmeasures.markdown]: string

File: source/_integrations/flexmeasures.markdown [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+  description: The maximum state of charge that the flexible energy asset is allowed to reach.
+  required: true
+  type: float
+{% endconfiguration %}
+
+{% include integrations/config_flow.md %}
+
+## Schedule Sensor
+
+The Flexmeasures Schedule sensor shows the values of the power draw or supply at the start of each interval. It contains a startdatetime, unit of measurement, and a device class as well. The schedule is automatically shifted up to be in line with the sensor resolution.
```
Comment (victorgarcia98) [inline source/_integrations/flexmeasures.markdown]: startdatetime -> start datetime

File: source/_integrations/flexmeasures.markdown [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+```
+
+## Automate scheduling
+
+The intended usage of FlexMeasures is letting the schedules and Home Assistant optimize the flexible energy assets without user interaction. Automations can be used to trigger new schedules when new information is available that would impact the schedule. Some examples of events that should trigger the request of new schedules are when the battery/asset is connected, when new prices are available, and periodically to match the executed schedule to the calculated schedule. The user will need to provide a Home Assistant entity as a `soc_at_start` to be used for triggering a schedule. This is an example `yaml` for the action set in the automations:
+
+```yaml
+service: flexmeasures.trigger_and_get_schedule
+data:
+  soc_at_start: "\{\{ state_attr('SENSOR_TYPE.SENSOR', 'SENSOR_ATTRIBUTES') }}"
```
Comment (victorgarcia98) [inline source/_integrations/flexmeasures.markdown:141]: is the escaping also needed for the closing pharenthesis?

**Review by victorgarcia98 — COMMENTED** [review]
Nice! Just some minor details.

Regarding the S2 I would add the following:

## S2 Protocol
The S2 Protocol is a flexibility description protocol that allows resource manager share their preferences with solutions that optimize energy assets (the Customer Energy Managers).

This integration allows to have FlexMeasures exposed as a Customer Energy Manager just in HomeAssistant. That way, users can change the control type of a resource manager in HA interface.


### Changing the control type 

A user can use the service `change_control_type` to change the control type of the resource manager, for example:

```yaml
service: flexmeasures.change_control_type
data:
  control_type: "FILL_RATE_BASED_CONTROL"
```

File: source/_integrations/flexmeasures.markdown [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+  unit: "SENSOR_UNIT_OF_THE_VALUES"
+  prior: "DATETIME_STRING_AT_WHICH_THE_VALUES_WERE_RECORDED_OR_FORECASTED"
+```
+
+##S2 Protocol
+
+The S2 Protocol is a flexibility description protocol that allows a resource manager share their preferences with solutions that optimize energy assets (the Customer Energy Managers).
+
+This integration allows to have FlexMeasures exposed as a Customer Energy Manager just in HomeAssistant. That way, users can change the control type of a resource manager in HA interface.
+Changing the control type
```
Comment (Flix6x) [inline source/_integrations/flexmeasures.markdown]: Besides my suggested rewrite in PR #2, this paragraph made me wonder whether our integration allows the S2 support to be toggled on or off (and if so, how), and what "changing the control type" is supposed to accomplish.

File: source/_integrations/flexmeasures.markdown (hunk 5 above)
Comment (GustaafL) [reply source/_integrations/flexmeasures.markdown:141]: I don't know how to escape this properly but only these are needed. I hope a reviewer will let us know how to do this properly or remove this section.

File: source/_integrations/flexmeasures.markdown (hunk 1 above)
Comment (GustaafL) [reply source/_integrations/flexmeasures.markdown]: Thanks, updated to The duration of the requested schedule in ....

File: source/_integrations/flexmeasures.markdown (hunk 2 above)
Comment (GustaafL) [reply source/_integrations/flexmeasures.markdown]: Fixed the type definition of the duration. Good catch

File: source/_integrations/flexmeasures.markdown (hunk 4 above)
Comment (GustaafL) [reply source/_integrations/flexmeasures.markdown]: Thanks for catching this typo

File: source/_integrations/flexmeasures.markdown (hunk 6 above)
Comment (GustaafL) [reply source/_integrations/flexmeasures.markdown]: maybe @victorgarcia98 knows but I can't explain it.

**Pull request merged**